File: code/string-algorithms/palindrome_checker/palindrome.rs
// Part of Cosmos by OpenGenus Foundation

fn is_palindrome(input: &str) -> bool {
    let reversed: String = input.chars().rev().collect();
    reversed == input
}

fn main() {
    println!("{:?}", is_palindrome("test"));
    println!("{:?}", is_palindrome("hahah"));
}

File: code/string_algorithms/src/palindrome_checker/palindrome.rb
# string algorithms | palindrome checker | palindrome | RUBY
// Part of Cosmos by OpenGenus Foundation

=end


# Part of Cosmos by OpenGenus Foundation
class String
  def palindrome?
    self == to_s.reverse
  end
end

puts 'test'.palindrome?
puts 'hahah'.palindrome?
